vortexor/src/vortexor.rs

```diff
@@ -177,3 +178,22 @@ impl Vortexor {
         Ok(())
     }
 }
+
+/// Binds the sockets to the specified address and port range if address is Some.
+/// If the address is None, it binds to the specified bind_address and port range.
+fn bind_sockets(
+    bind_address: std::net::IpAddr,
+    port_range: (u16, u16),
+    address: Option<SocketAddr>,
+    num_quic_endpoints: usize,
+    quic_config: SocketConfig,
+) -> Vec<UdpSocket> {
+    let (bind_address, port_range) = address
+        .map(|addr| (addr.ip(), (addr.port(), addr.port().saturating_add(1))))
+        .unwrap_or((bind_address, port_range));
+
+    let (_, socket) = bind_in_range_with_config(bind_address, port_range, quic_config)
```

Comment:
please use `multi_bind_in_range_with_config`, the pattern where `bind_in_range_with_config` is followed by `bind_more_with_config` is unsound and may result in binding on top of already bound ports.

Reply:
done